### PR TITLE
Add acl

### PR DESCRIPTION
**Pull request by dotChris90**
Specify library name and version: libacl/2.3.1

Libacl is a linux library for access control and used in the iceoryx project for PR https://github.com/conan-io/conan-center-index/pull/5541 .

- Just latest version
- using tools module over os module
- test_package simple and link against the CONAN_LIBS variable
- license written according to their homepage https://savannah.nongnu.org/projects/acl/

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — DISMISSED** [review]

File: recipes/libacl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+import os
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+class LibAclConan(ConanFile):
+    name = "libacl"
```
Comment (SpaceIm) [inline recipes/libacl/all/conanfile.py:8]:
```suggestion
    name = "libacl"
```

acl seems to be far more common name:
https://repology.org/project/acl/versions
https://repology.org/project/libacl/versions

File: recipes/libacl/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(
+            self, win_bash=tools.os_info.is_windows)
```
Comment (SpaceIm) [inline recipes/libacl/all/conanfile.py:56]:
```suggestion
        self._autotools = AutoToolsBuildEnvironment(self)
```

if windows is not supported, it's useless to add win_bash

File: recipes/libacl/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    def package(self):
+        with tools.chdir(self._source_subfolder):
+            autotools = self._configure_autotools()
+            autotools.install()
```
Comment (SpaceIm) [inline recipes/libacl/all/conanfile.py:79]:
```suggestion
        self._autotools.configure(configure_dir=self._source_subfolder, args=conf_args)
        return self._autotools

    def build(self):
        autotools = self._configure_autotools()
        autotools.make()

    def package(self):
        autotools = self._configure_autotools()
        autotools.install()
```

File: recipes/libacl/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        self.copy("COPYING", dst="licenses", src=self._doc_folder)
+        tools.rmdir(os.path.join(self.package_folder,"lib","pkgconfig"))
```
Comment (SpaceIm) [inline recipes/libacl/all/conanfile.py:81]:
```suggestion
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

File: recipes/libacl/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(example)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+
```
Comment (SpaceIm) [inline recipes/libacl/all/test_package/CMakeLists.txt:9]:
```suggestion
```

File: recipes/libacl/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            bin_path = os.path.join("bin", "example")
+            self.run(bin_path, run_environment=True)
+
```
Comment (SpaceIm) [inline recipes/libacl/all/test_package/conanfile.py:17]:
```suggestion
```

File: recipes/libacl/config.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+versions:
+  "2.3.1":
+    folder: all
+
```
Comment (SpaceIm) [inline recipes/libacl/config.yml:4]:
```suggestion
```

File: recipes/libacl/all/test_package/conanfile.py (hunk 6 above)
Comment (SpaceIm) [inline recipes/libacl/all/test_package/conanfile.py:17]:
```suggestion
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
Please apply Space's suggestions

File: recipes/libacl/all/conanfile.py (hunk 1 above)
Comment (dotChris90) [reply recipes/libacl/all/conanfile.py:8]: i would change the name to acl - shall i stop the PR then and start a new one so the folder can be changed from libacl to acl?

Comment (dotChris90) [reply recipes/libacl/all/conanfile.py:8]: i would first change the name in conanfile.py and you guys then can inform me if you would prefer renaming to folder too. i think for "clean style" i should change the foldername too

Comment (SpaceIm) [reply recipes/libacl/all/conanfile.py:8]: No need to open a new PR

Comment (dotChris90) [reply recipes/libacl/all/conanfile.py]: wow holy! :D i changed the foldername and jenkins still accept. 👍  
I was little worry that the servers maybe thinks that i push now a 2nd package in this PR. but nope - all ok. 
Cool - thanks!

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/acl/all/conandata.yml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "2.3.1":
+    url: "http://download.savannah.nongnu.org/releases/acl/acl-2.2.53.tar.gz"
+    sha256: "06BE9865C6F418D851FF4494E12406568353B891FFE1F596B34693C387AF26C7"
```
Comment (SpaceIm) [inline recipes/acl/all/conandata.yml:4]: wrong version

Comment (dotChris90) [reply recipes/acl/all/conandata.yml]: how embarrassing ..... ok i checked again all .... i hope this is now ok .... >.<

**Review by prince-chrismc — APPROVED** [review]

File: recipes/acl/all/conandata.yml (hunk 8 above)
Comment (SpaceIm) [reply recipes/acl/all/conandata.yml]: Why not download .tar.xz as I suggested? It's smaller.

Comment (dotChris90) [reply recipes/acl/all/conandata.yml]: though tar.gz is more common format and didnt saw that difference is so big ^^'

File: recipes/acl/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        conf_args = [
+            "--prefix={}".format(tools.unix_path(self.package_folder)),
+            "--bindir={}".format(tools.unix_path(
+                os.path.join(self.package_folder, "bin"))),
+            "--libdir={}".format(tools.unix_path(
+                os.path.join(self.package_folder, "lib")))
```
Comment (SpaceIm) [inline recipes/acl/all/conanfile.py:61]: Is it not already handled by `AutoToolsBuildEnvironment`?

Comment (dotChris90) [reply recipes/acl/all/conanfile.py:61]: hmmm awesome you are right. i saw multiple times in recipes and then though it is required.

File: recipes/acl/all/test_package/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(example)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
```
Comment (SpaceIm) [inline recipes/acl/all/test_package/CMakeLists.txt:7]:
```suggestion
project(example C)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup()

add_executable(${PROJECT_NAME} test_package.c)
```

Prefer C test_package for a C library.

File: recipes/acl/all/test_package/CMakeLists.txt [hunk 11]
```diff
@@ -1,8 +1,8 @@
 cmake_minimum_required(VERSION 3.1)
-project(example)
+project(example_c)
```
Comment (SpaceIm) [inline recipes/acl/all/test_package/CMakeLists.txt:2]:
```suggestion
project(example C)
```

File: recipes/acl/all/test_package/conanfile.py [hunk 12]
```diff
@@ -12,5 +12,5 @@ def build(self):
 
     def test(self):
         if not tools.cross_building(self.settings):
-            bin_path = os.path.join("bin", "example")
+            bin_path = os.path.join("bin", "example_c")
```
Comment (SpaceIm) [inline recipes/acl/all/test_package/conanfile.py:15]:
```suggestion
            bin_path = os.path.join("bin", "example")
```

**Review by prince-chrismc — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**